### PR TITLE
[WIP] CUDA 11.0 support

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

File: windows/internal/smoke_test.bat [hunk 1]
```diff
@@ -2,7 +2,7 @@ set SRC_DIR=%~dp0
 
 pushd %SRC_DIR%\..
 
-if "%CUDA_VERSION%" == "102" call internal\driver_update.bat
+if "%CUDA_VERSION%" == "110" call internal\driver_update.bat
```
Comment (peterjc123) [inline windows/internal/smoke_test.bat:5]: Maybe need a new line for this.

File: windows/internal/cuda_install.bat [hunk 2]
```diff
@@ -87,6 +88,22 @@ if not exist "%SRC_DIR%\temp_build\cudnn-10.2-windows10-x64-v7.6.5.32.zip" (
     set "CUDNN_SETUP_FILE=%SRC_DIR%\temp_build\cudnn-10.2-windows10-x64-v7.6.5.32.zip"
 )
 
+:cuda110
+
+if not exist "%SRC_DIR%\temp_build\cuda_11.0.1_451.22_win10.exe" (
+    curl -k -L http://developer.download.nvidia.com/compute/cuda/11.0.1/local_installers/cuda_11.0.1_451.22_win10.exe --output "%SRC_DIR%\temp_build\cuda_11.0.1_451.22_win10.exe"
```
Comment (peterjc123) [inline windows/internal/cuda_install.bat:94]: I can upload it to S3 if you want.

File: conda/build_pytorch.sh [hunk 3]
```diff
@@ -242,6 +242,9 @@ else
     . ./switch_cuda_version.sh "$desired_cuda"
     # TODO, simplify after anaconda fixes their cudatoolkit versioning inconsistency.
     # see: https://github.com/conda-forge/conda-forge.github.io/issues/687#issuecomment-460086164
+    if [[ "$desired_cuda" == "11.0" ]]; then
```
Comment (peterjc123) [inline conda/build_pytorch.sh:245]: https://anaconda.org/anaconda/cudatoolkit/files Looks like it has not be released in anaconda.

File: windows/internal/nightly_defaults.bat [hunk 4]
```diff
@@ -148,7 +148,7 @@ if "%PYTORCH_BUILD_VERSION%" == "" set PYTORCH_BUILD_VERSION=1.5.0.dev%NIGHTLIES
 
 if "%~1" == "Wheels" (
     if "%BUILD_PYTHONLESS%" == "" (
-        if not "%CUDA_VERSION%" == "102" (
+        if not "%CUDA_VERSION%" == "110" (
```
Comment (peterjc123) [inline windows/internal/nightly_defaults.bat:151]: So we use cuda 11 as default, cc @seemethere ?

File: windows/internal/cuda_install.bat [hunk 5]
```diff
@@ -87,6 +88,22 @@ if not exist "%SRC_DIR%\temp_build\cudnn-10.2-windows10-x64-v7.6.5.32.zip" (
...
+    if errorlevel 1 exit /b 1
+    set "CUDA_SETUP_FILE=%SRC_DIR%\temp_build\cuda_11.0.1_451.22_win10.exe"
+    set "ARGS=nvcc_11.0 cuobjdump_11.0 nvprune_11.0 cupti_11.0 cublas_11.0 cublas_dev_11.0 cudart_11.0 cufft_11.0 cufft_dev_11.0 curand_11.0 curand_dev_11.0 cusolver_11.0 cusolver_dev_11.0 cusparse_11.0 cusparse_dev_11.0 nvgraph_11.0 nvgraph_dev_11.0 npp_11.0 npp_dev_11.0 nvrtc_11.0 nvrtc_dev_11.0 nvml_dev_11.0"
+)
+
+REM cuDNN is not released yet
```
Comment (peterjc123) [inline windows/internal/cuda_install.bat:100]: cuDNN 8 RC is out.

File: packaging/pkg_helpers.bash [hunk 6]
```diff
@@ -52,6 +52,13 @@ setup_cuda() {
```
Comment (zasdfgbnm) [inline packaging/pkg_helpers.bash:52]: Also needs to change https://github.com/pytorch/builder/blob/master/packaging/README.md

File: windows/internal/smoke_test.bat (hunk 1 above)
Comment (zasdfgbnm) [reply windows/internal/smoke_test.bat:5]: fixed in https://github.com/pytorch/builder/pull/463

File: windows/internal/cuda_install.bat (hunk 2 above)
Comment (zasdfgbnm) [reply windows/internal/cuda_install.bat:94]: Feel free to do so if you prefer S3, and I will update the link here. Also, we just get a link for cuDNN as well, see: https://github.com/pytorch/builder/pull/463

Comment (zasdfgbnm) [reply windows/internal/cuda_install.bat:94]: http://developer.download.nvidia.com/compute/cuda/11.0.1/local_installers/cuda_11.0.1_451.22_win10.exe
https://developer.download.nvidia.com/compute/redist/cudnn/v8.0.0/cudnn-11.0-windows-x64-v8.0.0.180.zip

File: windows/internal/cuda_install.bat (hunk 5 above)
Comment (zasdfgbnm) [reply windows/internal/cuda_install.bat:100]: added

File: windows/internal/nightly_defaults.bat (hunk 4 above)
Comment (zasdfgbnm) [reply windows/internal/nightly_defaults.bat:151]: I reverted this file in https://github.com/pytorch/builder/pull/463
We could do the change late if we decide to.

**Pull request closed without merging**